### PR TITLE
removing member (global) variable cfg

### PR DESCRIPTION
**Pull request by tulinkry**
- most of the changes are just about introducing local `cfg` variable.
- on same places it was necessary to store the "context" into the request
- the most difficult was the `history.jsp` where there was an `if` statement from the top to the bottom - please check it

This completely solves the problem in #1392

### CODE REVIEW

File: web/history.jsp [hunk 1]
```diff
@@ -86,90 +65,149 @@ include file="httpheader.jspf"
...
+                    } else if (label == labelEnd && i < totalHits) {
+                        buf.append("&gt;&gt;");
+                    } else {
+                        buf.append(label);
+                    }
+                    buf.append("</a>");
+                }
+                label++;
+            }
+            slider = buf.toString();
```
Comment (vladak) [inline web/history.jsp]: seems like a candidate for getting this into method in `PageConfig` ?

Comment (tulinkry) [inline web/history.jsp]: Quite yes. It is already three times in the code

**Comment by vladak** [comment]
I am still wondering why the `private PageConfig cfg` in `pageconfig.jspf` did not work.

File: web/history.jsp [hunk 2]
```diff
@@ -86,90 +65,149 @@ include file="httpheader.jspf"
...
+            request.setAttribute("history.jsp-slider", slider);
+        } else {
+            // set the max index to max or last
+            thispage = totalHits - start;
+        }
+%>
         </div>
         <div id="Masthead">History log of 
         <%= Util.breadcrumbPath(context + Prefix.XREF_P, path,'/',"",true,cfg.isDir()) %>
```
Comment (vladak) [inline web/history.jsp]: what about this `cfg` ?

**Comment by vladak** [comment]
In `history.jsp` there are bunch of `cfg` references (`cfg.isDir()` calls on L234, L272, L285 + `cfg.getRevisionMessageCollapseThreshold()` on L359) where I am not sure about `cfg` origin.

**Comment by vladak** [comment]
Other than the comments above this looks good.

**Comment by vladak** [comment]
Just an idea for `history.jsp`: try to get it compiled into Java code and then search for all `cfg` occurrences.

File: web/history.jsp (hunk 2 above)
Comment (tulinkry) [inline web/history.jsp]: In scope from line 95. The scope ends on 161.

**Comment by tulinkry** [comment]
> In history.jsp there are bunch of cfg references (cfg.isDir() calls on L234, L272, L285 + cfg.getRevisionMessageCollapseThreshold() on L359) where I am not sure about cfg origin.

The local scope for this `cfg` is 173-434

**Comment by tulinkry** [comment]
> Just an idea for history.jsp: try to get it compiled into Java code and then search for all cfg occurrences.

That's what I was doing the whole time to fix the missing member variable.

**Comment by tulinkry** [comment]
> I am still wondering why the private PageConfig cfg in pageconfig.jspf did not work.

[As you found in the `PageConfig`](https://github.com/OpenGrok/OpenGrok/issues/1398#issuecomment-280021861) the page config must not be shared among requests. And this member variable **was** shared among parallel requests because the jsp is turned into a simple servlet and tomcat holds only one instance of a servlet to serve all incoming requests.

File: web/history.jsp (hunk 1 above)
Comment (tulinkry) [inline web/history.jsp]: I extracted the slider into utils

**Comment by vladak** [comment]
Looks good, pls merge.

**Pull request merged**